comp/otelcol/collector/impl-pipeline/flare_filler.go

```diff
@@ -0,0 +1,149 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build otlp
+
+// Package collector implements the collector component
+package collector
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	flaretypes "github.com/DataDog/datadog-agent/comp/core/flare/types"
+	extension "github.com/DataDog/datadog-agent/comp/otelcol/extension/def"
+	"github.com/gocolly/colly/v2"
+)
+
+func (c *collectorImpl) fillFlare(fb flaretypes.FlareBuilder) error {
+	if !c.config.GetBool("otel.enabled") {
+		fb.AddFile("otel/otel-agent.log", []byte("'otel.enabled' is disabled in the configuration"))
+		return nil
+	}
+
+	// request config from Otel-Agent
+	responseBytes, err := c.requestOtelConfigInfo(c.config.GetString("otel.extension_url"))
+	if err != nil {
+		fb.AddFile("otel/otel-agent.log", []byte(fmt.Sprintf("did not get otel-agent configuration: %v", err)))
+		return nil
+	}
```

Comment:
Can we log error ?

Reply:
I _think_ this bubbles up...

Reply:
From testing, that doesn't seem to be the case that it bubbles up. I'm adding a log.Error here as it will only be hit if the OTel Collector is enabled but isn't reachable.